test/standalone/test-debuglet.js

```diff
@@ -195,11 +205,88 @@ describe(__filename, function(){
     debuglet.start();
   });
 
-  it('should de-activate when the server responds with isDisabled');
+  it('should de-activate when the server responds with isDisabled', function(done) {
+    this.timeout(4000);
+    var config = extend({}, defaultConfig, {projectId: 'fake-project'});
+    debuglet = new Debuglet(fakeDebug, config, logger);
+
+    var scope = nock(API)
+      .post(REGISTER_PATH)
+      .reply(200, {
+        debuggee: {
+          id: DEBUGGEE_ID,
+          isDisabled: true
+        }
+      })
+      .post(REGISTER_PATH)
+      .reply(200, {
+        debuggee: {
+          id: DEBUGGEE_ID
+        }
+      });
+
+    debuglet.once('registered', function(id) {
+      assert(id === DEBUGGEE_ID);
+      scope.done();
+      done();
+    });
+
+    debuglet.start();
+  });
+
+  it('should re-register when registration expires', function(done) {
```

Comment:
The test doesn't seem to be doing what was advertised. This is testing whether we retry registration upon a 4xx error (specifically 404).

Registration is expired by a 404 response on the `breakpoints.list` API. See [here](https://github.com/GoogleCloudPlatform/cloud-debug-nodejs/blob/c580792c1031f3fb51fdf9dc03ef1738db86f46c/src/agent/debuglet.js#L286).